From 89ac6b353521fb9a6d6eb6bb74724c4fa968f75c Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Sat, 17 Dec 2022 18:17:33 +0100
Subject: [PATCH] =?UTF-8?q?alsa:=20don=C2=B4t=20set=20-1=20as=20node.targe?=
 =?UTF-8?q?t?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Leave the target unset when -1, just like what pw-stream does.

See #2893
---
 pipewire-alsa/alsa-plugins/pcm_pipewire.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pipewire-alsa/alsa-plugins/pcm_pipewire.c b/pipewire-alsa/alsa-plugins/pcm_pipewire.c
index 37a907b24..96b57da78 100644
--- a/pipewire-alsa/alsa-plugins/pcm_pipewire.c
+++ b/pipewire-alsa/alsa-plugins/pcm_pipewire.c
@@ -579,7 +579,7 @@ static int snd_pcm_pipewire_prepare(snd_pcm_ioplug_t *io)
 		pw_properties_setf(props, PW_KEY_NODE_LATENCY, "%lu/%u", pw->min_avail, io->rate);
 	if (pw_properties_get(props, PW_KEY_NODE_RATE) == NULL)
 		pw_properties_setf(props, PW_KEY_NODE_RATE, "1/%u", io->rate);
-	if (pw->target != NULL &&
+	if (pw->target != NULL && !spa_streq(pw->target, "-1") &&
 		pw_properties_get(props, PW_KEY_NODE_TARGET) == NULL)
 		pw_properties_setf(props, PW_KEY_NODE_TARGET, "%s", pw->target);
 
